src/api_2.rs

```diff
@@ -0,0 +1,207 @@
+						offset,
+						app_id,
+						block_num
+					);
+					generate_app_specific_cells(*offset, max_cols, block_num, block, *app_id)
+				},
+				None => {
+					vec![]
+				},
+			};
+			Ok(cells)
+		}
+
+		fn mk_response(s: String) -> Result<Response<Body>, hyper::Error> {
+			Ok(Response::builder()
+				.status(200)
+				.header(ACCESS_CONTROL_ALLOW_ORIGIN, "*")
+				.header("Content-Type", "application/json")
+				.body(Body::from(s))
+				.unwrap())
+		}
+
+		let local_tm: DateTime<Local> = Local::now();
+		log::info!(
+			"⚡️ {} | {} | {}",
+			local_tm.to_rfc2822(),
+			req.method(),
+			req.uri().path()
+		);
+
+		let _db = self.store.clone();
+		let url = self.url.clone();
+
+		Box::pin(async move {
+			let res = match req.method() {
+				&Method::GET => {
+					if let Ok(block_num) = match_url_block(req.uri().path()) {
+						if let Ok(app_id) = match_url_id(req.uri().path()) {
+							//cell logic to be written here
+							let block = get_block_by_number(&url, block_num).await.unwrap();
+							let header = block.clone().header;
+							// let max_rows = header.extrinsics_root.rows;
+							// let max_cols = header.extrinsics_root.cols;
+
+							// let cells = get_kate_proof(&url, block_num, max_rows, max_cols, app_id as u32).await.unwrap();
+							let mut vec = Vec::new();
+							// for cell in cells.iter(){
+							//     vec.push((cell.proof).clone());
+							// }
+							let cells =
+								check_id(header, app_id as u32, block_num, block.clone()).unwrap();
+							for i in cells.iter() {
+								let data =
+									get_kate_query_proof_by_cell(&url, block_num, i.row, i.col)
+										.await
+										.unwrap();
+								vec.push(data);
+							}
+							mk_response(
+								format!(r#"{{"block": {}, "data": {:?}}}"#, block_num, vec)
+									.to_owned(),
+							)
+						} else {
+							let mut not_found = Response::default();
+							*not_found.status_mut() = StatusCode::NOT_FOUND;
+							not_found
+								.headers_mut()
+								.insert(ACCESS_CONTROL_ALLOW_ORIGIN, "*".parse().unwrap());
+							Ok(not_found)
+						}
+					} else {
+						let mut not_found = Response::default();
+						*not_found.status_mut() = StatusCode::NOT_FOUND;
+						not_found
+							.headers_mut()
+							.insert(ACCESS_CONTROL_ALLOW_ORIGIN, "*".parse().unwrap());
+						Ok(not_found)
+					}
+				},
+				_ => {
+					let mut not_found = Response::default();
+					*not_found.status_mut() = StatusCode::NOT_FOUND;
+					not_found
+						.headers_mut()
+						.insert(ACCESS_CONTROL_ALLOW_ORIGIN, "*".parse().unwrap());
+					Ok(not_found)
+				},
+			};
+			res
+		})
+	}
+}
+
+struct MakeHandler {
+	store: Arc<DBWithThreadMode<SingleThreaded>>,
+	url: String,
+}
+
+impl<T> Service<T> for MakeHandler {
+	type Error = hyper::Error;
+	type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
+	type Response = Handler;
+
+	fn poll_ready(&mut self, _: &mut Context) -> Poll<Result<(), Self::Error>> {
```

Comment:
Unsure if we really need this function, we could just call the `Poll::Ready` directly.

Reply:
same in the case if poll_ready()